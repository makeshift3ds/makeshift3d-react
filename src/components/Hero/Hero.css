@import '../../assets/css/globals.css';

.hero {
  height: 300px;
  background-image: url('../../assets/img/hero-background.svg');
  background-repeat: no-repeat;
  background-size: cover;
  background-position: top right;
  background-color: var(--tan);

  & .hero-container {
    max-width: var(--max-width);
    margin: 0 auto;
  }

  & .avatar {
    position: relative;
    margin-top: 225px;
    margin-left: 25px;
    height: 150px;
    width: 150px;
    border-radius: 50%;
    border: 3px solid var(--white);
    z-index: 10;
    background-image: url(//res.cloudinary.com/makeshift3ds/image/upload/makeshift-header-avatar_omdbug.jpg);
    background-size: cover;
    background-position: center center;

    @media (--small) {
      position: static;
      margin: 0 auto;
      margin-top: calc((300px/2) - (150px/2));
    }
  }
}