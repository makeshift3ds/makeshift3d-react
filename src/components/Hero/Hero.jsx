import React from 'react';

const Hero = () => (
  <section className="hero">
    <div className="hero-container">
      <div className="avatar" />
    </div>
  </section>
);

export default Hero;
