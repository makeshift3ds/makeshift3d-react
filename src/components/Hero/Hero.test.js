import React from 'react';
import { shallow } from 'enzyme';
import Hero from './Hero.jsx';

describe('<Hero />', () => {
  it('should render 1 <Hero /> component', () => {
    const wrapper = shallow(<Hero />);
    expect(wrapper).toHaveLength(1);
  });

  it('should render an avatar element', () => {
    const component = shallow(<Hero />);
    expect(component.find('.avatar')).toHaveLength(1);
  });
});
