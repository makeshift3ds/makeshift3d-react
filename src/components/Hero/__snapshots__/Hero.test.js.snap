// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Hero /> should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Hero />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="hero-container"
      >
        <div
          className="avatar"
        />
      </div>,
      "className": "hero",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="avatar"
        />,
        "className": "hero-container",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "avatar",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      "type": "div",
    },
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="hero-container"
        >
          <div
            className="avatar"
          />
        </div>,
        "className": "hero",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="avatar"
          />,
          "className": "hero-container",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "avatar",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        "type": "div",
      },
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
